libs/designsystem/src/lib/components/list/list.component.ts

```diff
@@ -116,21 +116,21 @@ export class ListComponent implements OnInit, AfterViewInit, OnChanges {
   };
 
   private async getVirtualDataset(index: number, count: number): Promise<any> {
-    const slicePromise = await new Promise((resolve) => {
+    return await new Promise((resolve) => {
       setTimeout(() => {
         const sliceWithMeta = this.getItemsSliceWitMeta(index, count);
 
         // If we return less items than count, virtual scroll will interprete it as EOF and stop asking for more
         if (sliceWithMeta.length < count && this.isLoadOnDemandEnabled) {
           let elapsedTime = 0;
           // Scrollend (that triggers load on demand) is not fired when we scroll as the virtual
```

Comment:
Tought this comment was a bit verbose, so I took a stab at it :) 
```
/* As virtual scroll fixes the viewport causing ScrollEnd to not be emitted; do it manually to trigger load on demand */ 
*/
```

I don't know if it is much better... 🤷‍♂️

Reply:
Better